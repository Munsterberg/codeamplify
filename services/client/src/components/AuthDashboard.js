import React, { Component } from 'react';
import { object } from 'prop-types';

import AuthForm from './AuthForm';

class AuthDashboard extends Component {
  state = {
    formData: {
      username: '',
      email: '',
      password: ''
    }
  }

  static propTypes = {
    location: object,
  }

  render() {
    const { formData } = this.state;
    const formType = this.props.location.pathname === '/register' ? 'Register' : 'Login';
    return (
      <AuthForm formData={formData} formType={formType} />
    );
  }
}

export default AuthDashboard;