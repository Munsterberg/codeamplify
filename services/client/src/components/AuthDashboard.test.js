import React from 'react';
import { shallow } from 'enzyme';

import AuthDashboard from './AuthDashboard';

describe('<AuthDashboard />', () => {
  let wrapper;
  const location = {
    pathname: '/register'
  };

  beforeEach(() => {
    wrapper = shallow(<AuthDashboard location={location} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });
});