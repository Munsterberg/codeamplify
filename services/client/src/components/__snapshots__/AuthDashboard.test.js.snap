// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthDashboard /> should render correctly 1`] = `
<AuthForm
  formData={
    Object {
      "email": "",
      "password": "",
      "username": "",
    }
  }
  formType="Register"
/>
`;
